### ISSUE TITLE
adding guards exposed that the map code was not authenticating

### ISSUE THREAD

**Issue by shankari** [NREL/crowdsource-mask-wearing#9]
As part of https://github.com/NREL/crowdsource-mask-wearing/pull/7, I correctly implemented auth guards for the client and included the auth handler for the server. After this, the map call started failing. This is because the `maskreports` call from the map does not include the auth token.

I verified this even in the client without the changes

### Call from reports

```
Host: localhost:8080
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:68.0) Gecko/20100101 Firefox/68.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Authorization: Bearer <redacted>
DNT: 1
Connection: keep-alive
Referer: http://localhost:8080/reports
Cookie: <redacted>
If-None-Match: W/"e81d5-sMDLU9PQYLlfu5rBVjUmAbUgyfU"
```

#### Call from map

```
Host: localhost:8080
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:68.0) Gecko/20100101 Firefox/68.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
DNT: 1
Connection: keep-alive
Referer: http://localhost:8080/map
Cookie: <redacted>
If-None-Match: W/"e81d5-sMDLU9PQYLlfu5rBVjUmAbUgyfU"
```

This didn't fail before because we were not checking for the auth token on the server.

**Comment by shankari** [comment]
I am not quite sure why that is. We add the token through a http interceptor (`frontend/src/app/core/helpers/jwt.interceptor.ts`) so it should get added to all http calls if the token exists.

@fua0004673 I looked at the code and you make a separate http call instead of reusing the call from `maskreport.service`. And you use promises instead of observers. Do you have a specific reason why? I wonder if the different calling mechanism is the reason that the interceptor is not invoked. Thoughts?

**Comment by shankari** [comment]
I do see this note in the [interceptor documentation](https://angular.io/api/common/http/HttpInterceptor)

> To use the same instance of HttpInterceptors for the entire app, import the HttpClientModule only in your AppModule, and add the interceptors to the root application injector . If you import HttpClientModule multiple times across different modules (for example, in lazy loading modules), each import creates a new copy of the HttpClientModule, which overwrites the interceptors provided in the root module.

But that doesn't explain to me why it works for the report screen and not the map screen given that they are both in the same module.

**Comment by shankari** [comment]
Ah but maybe they are not in the same module? `frontend/src/app/feature/maskreport/maskreport.module.ts` doesn't include   `map.service.ts` or `map-data.service.ts`

**Comment by shankari** [comment]
I can confirm that the interceptor is not called for the map route. I don't know whether it only works on observables or whether it is related to the module loading. Moving on to finishing my own work first.

**Comment by fua0004673** [comment]
@shankari promises were used in this case because we only have a one time on page load get of the data which didn’t necessitate the use of observables at the time. Now that we’re doing multi user single db I think you are correct in that it makes more sense to be using observables to ensure that data properly stays in sync. I will work on making this change.

Regardless, the difference between observables and promises shouldn’t lead to a difference with an http interceptor - we’ve used apps that have utilized both together. The issue you mentioned with separate modules may be more likely - we will investigate more tonight.

**Comment by shankari** [comment]
Fixed by #11
